doc/news/_preparation_next_release.md

```diff
@@ -188,6 +188,12 @@ Thanks to Daniel Bugl.
   commonly used specification keywords to be used with the [typechecker](https://www.libelektra.org/plugins/typechecker). Currently the
   keywords `check/range`, `check/enum` and `check/validation` are supported. *(Armin Wurzinger)*
 
+### Typechecker
+
+- The plugin [typechecker](https://www.libelektra.org/plugins/typechecker) has been improved under the hood. It now supports a more concise
+and efficient typechecking process including a greatly improved type inference scheme that allows generate specification files and thus
+generated errors to be easier to understand. *(Armin Wurzinger)*
```

Comment:
Ülease give an example of such a better error message.

Reply:
in the readme now, explained how to interprete it there as well, good point